torch/nn/functional.py

```diff
@@ -2737,6 +3012,11 @@ def grid_sample(input, grid, mode='bilinear', padding_mode='zeros', align_corner
         Since then, the default behavior has been changed to ``align_corners = False``,
         in order to bring it in line with the default for :func:`interpolate`.
     """
+    tens_ops = (input, grid)
```

Comment:
For my curiosity; how come in some of the adjusted code, we need to define `tens_ops`, and in other cases, we don't? Is this just a code reduction measure?

Reply:
If there's multiple tensor operands I create the tuple before calling `type` because I need to create the `tuple` no matter what. When there's only one tensor operand I don't create the tuple because it's only needed if `type(input)` ends up not being `Tensor`, so we don't need to pay the cost of creating the tuple unless we're passed non-tensor operands. I figure optimizing for tensor operands is what you'd prefer :)